torch/_functorch/_aot_autograd/jit_compile_runtime_wrappers.py

```diff
@@ -792,6 +795,329 @@ def propagate_meta_info(new_hop_gm, new_call_function_node, old_call_function_no
+        fw_g_inputs = list(fw_g.find_nodes(op="placeholder"))
+        for idx in static_input_indices:
+            static_inputs.add(fw_g_inputs[idx])
+
+    for saved in fw_outs_saved_for_bw:
+        if saved in static_inputs:
+            # Do not run hooks on static_input_indices,
+            # as they correspond to model parameters,
+            # which have more users than autograd save context.
+            # The main use case for saved_tensors_hooks is activation quantization.
+            # Where desired behavior is to quantize saved activations
+            # to free the original saved tensor.
+            # That can not be done for parameters (static_input_indices).
+            fw_outs_saved_tensors_unchanged.append(saved)
+            continue
+
+        val = saved.meta["val"]
+        if not isinstance(val, torch.Tensor):
+            continue
+
+        pack_out_val = pack_hook_gm(val)
+
+        requires_sc_handling = any(
+            is_traceable_wrapper_subclass(x) for x in pytree.tree_leaves(pack_out_val)
+        )
+        if requires_sc_handling:
+            raise NotImplementedError(
+                "Tensor subclasses in GraphModule saved tensors hooks are not supported"
+                "You can workaround it by manually returning subclass's inner tensors"
+                " in the pack hook, and reconstructing the subclass in the unpack hook"
+            )
+
+        pack_gm = prepare_hook_gm(aot_config, pack_hook_gm, (val,))
+        pack_g = pack_gm.graph
+        maybe_log_graph(
+            pack_gm,
+            f"saved_tensors_pack_hook {saved.name}",
+            aot_config,
+            lambda: f"aot_saved_tensors_hooks_pack {saved.name}",
+            structured_logs,
+        )
+        pack_out_val = pack_gm(val)
+
+        # Install pack hook graph as eiplogue of fw_module.
+        # Saved tensor output becomes input of pack hook graph.
+        # Replace saved tensor output with pack hook graph output.
+        # Outputs symbolic scalars, tensors  are accumulated separately.
+        # Then in forward outputs and backward inputs installed in order
+        # sym_scalars, packed_saved_tensors.
+        # Keeping all tensors together allows to preserve
+        # the same identification at runtime,
+        # updating only number of saved sym_scalars and tensors.
+        pack_g_inputs = pack_g.find_nodes(op="placeholder")
+        assert len(pack_g_inputs) == 1
+        env = {pack_g_inputs[0]: saved}
+        fw_pack_out_n = None
+        with fw_g.inserting_before(fw_out_n):
+            for node in pack_g.nodes:
+                if node.op == "placeholder":
+                    continue
+                new_n = fw_g.node_copy(node, lambda n: env[n])
+                env[node] = new_n
+                # Output node is temporarily copied to have remapped arguments.
+                # Removed in the end.
+                if node.op == "output":
+                    fw_pack_out_n = new_n
```

Comment:
nit: reading FX graph manipulation is always a bit painful to make this a bit clearer, you can set this to `fw_pack_out_args = `new_n.args[0]` directly

(I was reading the for loop a few lines below and it wasn't clear why you were iterating over `args[0]` - then i realized this is coming from an output node in the FX graph, which has this invariant)